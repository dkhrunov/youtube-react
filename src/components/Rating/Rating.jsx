import React from 'react';
import { Icon, Progress } from 'semantic-ui-react';
import './Rating.scss';

function buildProgressBar(likeCount, dislikeCount) {
	const percent = 100 * (likeCount / (dislikeCount + likeCount));
	return <Progress className='progress' percent={percent} size='tiny' />;
}

const Rating = (props) => {
	const { likeCount, dislikeCount } = props;

	let progress = null;

	if (likeCount && dislikeCount) {
		progress = buildProgressBar(likeCount, dislikeCount);
	}

	return (
		<div className='rating'>
			<div className='thumbs-up'>
				<Icon name='thumbs outline up'></Icon>
				<span>{likeCount}</span>
			</div>
			<div className='thumbs-down'>
				<Icon name='thumbs outline down'></Icon>
				<span>{dislikeCount}</span>
			</div>
			{progress}
		</div>
	);
};

export default Rating;
