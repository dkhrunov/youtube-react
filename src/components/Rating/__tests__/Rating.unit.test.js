import React from 'react';
import { shallow } from 'enzyme';
import Rating from '../Rating';

describe('Renders Rating component', () => {
	test('renders without props', () => {
		const wrapper = shallow(<Rating />);
		expect(wrapper).toMatchSnapshot();
	});

	test('renders with likeCount props', () => {
		const wrapper = shallow(<Rating likeCount={123} />);
		expect(wrapper).toMatchSnapshot();
	});

	test('renders with dislikeCount props', () => {
		const wrapper = shallow(<Rating dislikeCount={234} />);
		expect(wrapper).toMatchSnapshot();
	});

	test('renders with likeCount and dislikeCount props', () => {
		const wrapper = shallow(<Rating likeCount={1003} dislikeCount={656} />);
		expect(wrapper).toMatchSnapshot();
	});
});
