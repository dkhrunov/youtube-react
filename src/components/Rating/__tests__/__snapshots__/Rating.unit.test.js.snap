// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Rating component renders with dislikeCount props 1`] = `
<div
  className="rating"
>
  <div
    className="thumbs-up"
  >
    <Icon
      as="i"
      name="thumbs outline up"
    />
    <span />
  </div>
  <div
    className="thumbs-down"
  >
    <Icon
      as="i"
      name="thumbs outline down"
    />
    <span>
      234
    </span>
  </div>
</div>
`;

exports[`Renders Rating component renders with likeCount and dislikeCount props 1`] = `
<div
  className="rating"
>
  <div
    className="thumbs-up"
  >
    <Icon
      as="i"
      name="thumbs outline up"
    />
    <span>
      1003
    </span>
  </div>
  <div
    className="thumbs-down"
  >
    <Icon
      as="i"
      name="thumbs outline down"
    />
    <span>
      656
    </span>
  </div>
  <Progress
    className="progress"
    percent={60.458107293550334}
    size="tiny"
  />
</div>
`;

exports[`Renders Rating component renders with likeCount props 1`] = `
<div
  className="rating"
>
  <div
    className="thumbs-up"
  >
    <Icon
      as="i"
      name="thumbs outline up"
    />
    <span>
      123
    </span>
  </div>
  <div
    className="thumbs-down"
  >
    <Icon
      as="i"
      name="thumbs outline down"
    />
    <span />
  </div>
</div>
`;

exports[`Renders Rating component renders without props 1`] = `
<div
  className="rating"
>
  <div
    className="thumbs-up"
  >
    <Icon
      as="i"
      name="thumbs outline up"
    />
    <span />
  </div>
  <div
    className="thumbs-down"
  >
    <Icon
      as="i"
      name="thumbs outline down"
    />
    <span />
  </div>
</div>
`;
